docs/tutorial/tutorial-5/android.rst

```diff
@@ -2,9 +2,382 @@
 Tutorial 5 - Taking it mobile: Android
 ======================================
 
-To deploy our applicaton to Android...
+The process of deploying an application to Android is very similar to the
+process for deploying as a desktop application. There is one additional
+pre-requisite: Java 8.
+
+Setting up Java 8
+=================
+
+To use BeeWare's Android tooling, you must install a Java 8 compiler and provide
+the path to it in the ``JAVA_HOME`` environment variable. This is because
+BeeWare's Android tooling depends on the Android SDK, and the Android SDK
+depends on the Java runtime and compiler being available. Some tools within
+the Android SDK specifically require version 8 of Java, so we stick to that
+in this documentation.
+
+Download & install Java 8
+-------------------------
+
+Visit the `AdoptOpenJDK website <https://adoptopenjdk.net/>`_ and
+download OpenJDK 8.
+
+.. tabs::
+
+  .. group-tab:: macOS
+
+    On macOS, the default format is a *.pkg* installer. Download and execute the
+    installer, accepting the defaults. In your terminal, run:
+
+    ``export JAVA_HOME="$(/usr/libexec/java_home -v 1.8)"``
+
+    You will need to run this every time you start a terminal session, or you
+    can add it to your `shell configuration <https://apple.stackexchange.com/a/356455>`_
+    or `launchd configuration <http://www.dowdandassociates.com/blog/content/howto-set-an-environment-variable-in-mac-os-x-launchd-plist/>`_.
```

Comment:
Really? Is adding it your launchd plist actually recommended? 

Regardless - preference is always to point at authoritative sources, rather than random sites on the net. Stackoverflow will do in a pinch; however, the complication here is that macOS is currently transitioning from bash to zsh, so there are 2 sets of instructions needed here. Can we find a set of instructions that are either on the AdoptOpenJDK (or Oracle) page for how to do this?

Reply:
I think it's fun to talk about neat ways to set environment variables! No reason the launchd plist is a bad idea that I can think of, and it sidesteps the bash vs. zsh difference.

AdoptOpenJDK doesn't have any such instructions I can find; see e.g. https://www.google.com/search?q=site%3Aadoptopenjdk.net+java_home&oq=site%3Aadoptopenjdk.net+java_home&aqs=chrome..69i57j69i58.7782j0j9&sourceid=chrome&ie=UTF-8

Oracle doesn't have any decent instructions that I can find. https://docs.oracle.com/javase/9/install/installation-jdk-and-jre-macos.htm#JSJIG-GUID-F575EB4A-70D3-4AB4-A20E-DBE95171AB5F doesn't really help. https://www.oracle.com/webfolder/technetwork/tutorials/oraclecode/mac-hol-setup.pdf exists but (1) is a PDF, and (2) isn't specific enough.

I think that given the options I found in ~10 min of research, the apple.stackexchange.com document is the best I found by far.

I'm not opposed to writing something myself here.

I'm open to discussion of the best way to handle this. Perhaps -> our call.

This all gets really complicated really fast because might be using terminal emulators which run their shell as non-login shell (which for bash changes which config files it reads), etc., which is why I like the launchd approach.